docs/nodes/metrics.md

```diff
@@ -18,40 +18,53 @@ Listen address can be changed in the config file (see
+| consensus_latest_block_height           | gauge     |                 | /status sync_info number                                                                                                                   |
+| consensus_fast_syncing                  | gauge     |                 | either 0 (not fast syncing) or 1 (syncing)                                                                                                 |
+| consensus_state_syncing                 | gauge     |                 | either 0 (not state syncing) or 1 (syncing)                                                                                                |
+| consensus_block_size_bytes              | Gauge     |                 | Block size in bytes                                                                                                                        |
+| consensus_step_duration                 | Histogram | step            | Histogram of durations for each step in the consensus protocol                                                                             |
+| consensus_block_gossip_receive_latency  | Histogram |                 | Histogram of time taken to receive a block in seconds, measure between when a new block is first discovered to when the block is completed |
+| consensus_block_gossip_parts_received   | Counter   | matches_current | Number of block parts received by the node                                                                                                 |
+| consensus_quorum_prevote_delay          | Gauge     |                 | Interval in seconds between the proposal timestamp and the timestamp of the earliest prevote that achieved a quorum                        |
+| consensus_full_prevote_delay            | Gauge     |                 | Interval in seconds between the proposal timestamp and the timestamp of the latest prevote in a round where all validators voted           |
+| consensus_proposal_timestamp_difference | Histogram |                 | Difference between the timestamp in the proposal message and the local time of the validator at the time it received the message           |
+| consensus_vote_extension_receive_count  | Counter   | status          | Number of vote extensions received                                                                                                         |
+| consensus_proposal_receive_count        | Counter   | status          | Total number of proposals received by the node since process start                                                                         |
+| consensus_proposal_create_count         | Counter   |                 | Total number of proposals created by the node since process start                                                                          |
```

Comment:
Wild idea: would the ratio between number of tx passed to App (`RequestPrepareProposal`) and the number of tx returned by the App (`ResponsePrepareProposal`) be a meaningful metric?

Reply:
Metrics are often helpful for diagnostic purposes: they help operators and developers detect deviation for normal behavior and understand what may be going wrong. I'm not sure I yet see what case that case that value may help operators detect.

Reply:
It could help detect an Application adding too many, or removing too many txs. So would be info mainly for App devs.
But it's a fair point this might not be the most important think to look at. I imagine the usual "rule of thumb" decide on a particular metric here is that, if it's not super important, it's probably OK to look into the Tendermint logs when this (unlikely case) happens.

Reply:
Interesting. I suppose that I think that any notion of 'too many' is so dependent on the contents of the mempool and the applications expected behavior. It may really be the right thing for applications to do to accept very little of the mempool generated txs in exchange for many application-generated txs or vice versa. A shift in the trend of that variable, either towards more application tx or more mempool tx, would be hard to use to diagnose an issue. Is the mempool suddenly receiving more garbage? Is the application in a state that needs to issue a lot of Tx?